msccl/language/mscclpp/__init__.py

```diff
@@ -189,6 +189,10 @@ def _get_buffer_index(self, remote_rank, buffer, index):
 
     def _put(self, dst, buffer=None, index=-1, sendtb=-1, chan_type=ChannelType.sm, use_packet=False):
         self.prog.check_buffer_exists(dst, buffer)
+        temp_chunk=None
+        if chan_type == ChannelType.proxy:
+            self.prog.check_buffer_exists(self.rank, "scratch")
```

Comment:
I think user need to provide the scratch buffer here. Otherwise, it may cause some unexpected behavior, such as different op use same scratch buffer